src/line_ending.rs

```diff
@@ -0,0 +1,117 @@
+//! Line ending detection and conversion.
+
+use std::fmt::{Debug, Formatter};
+use std::str::FromStr;
+
+/// Supported line endings. Like in the Rust's standard library, two
+/// line endings are supported: `\r\n` and `\n`
+#[derive(Clone, Copy, Debug, PartialEq)]
+pub enum LineEnding {
+    /// _Carriage return and line feed_ – a line ending sequence
+    /// historically used in _Windows_. Corresponds to the sequence
```

Comment:
I don't think we have a convention of putting product names in italic in the documentation, so please change `_Windows_` to `Windows` (and below for Unix).